### PR TITLE
Reinit binary addon manager when loading a profile; fixes non working pvr addons.

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

This is a fix where @ksooo helped and to have asap before a bigger fix and rework is approved.
See https://github.com/xbmc/xbmc/pull/18057

Thanks a lot @ksooo for this code.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by ksooo — APPROVED** [review]
Runtime-tested. PVR addons work again after switching profiles.

**Review by phunkyfish — APPROVED** [review]
👍

File: xbmc/addons/binary-addons/BinaryAddonManager.h [hunk 1]
```diff
@@ -32,6 +32,7 @@ namespace ADDON
     CBinaryAddonManager(const CBinaryAddonManager&) = delete;
     ~CBinaryAddonManager();
 
+    bool ReInit() { DeInit(); return Init(); }
```
Comment (Rechi) [inline xbmc/addons/binary-addons/BinaryAddonManager.h:35]: > Do not put multiple statements on a single line. Always use a new line for a new statement.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#41-multiple-statements

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

File: xbmc/profiles/ProfileManager.cpp [hunk 2]
```diff
@@ -374,6 +375,7 @@ void CProfileManager::FinalizeLoadProfile()
   PVR::CPVRManager &pvrManager = CServiceBroker::GetPVRManager();
   CNetworkBase &networkManager = CServiceBroker::GetNetwork();
   ADDON::CAddonMgr &addonManager = CServiceBroker::GetAddonMgr();
+  ADDON::CBinaryAddonManager &binAddonManager = CServiceBroker::GetBinaryAddonManager();
```
Comment (Rechi) [inline xbmc/profiles/ProfileManager.cpp:378]: > Left-align `*` and `&` to the base type they modify.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types

**Pull request merged**